packages/flutter/lib/src/rendering/mouse_cursor.dart

```diff
@@ -0,0 +1,270 @@
+/// [SystemMouseCursors].
+///
+/// [NoopMouseCursor] are prepared cursors that do nothing when switched onto. It
+/// is useful in special cases such as a platform view where the mouse cursor is
+/// managed by other means. Its singleton instance is available at
+/// [SystemMouseCursors.uncontrolled].
+///
+/// ## Using cursors
+///
+/// A [MouseCursor] object is used by being assigned to a [MouseRegion]. Many
+/// other widgets that use [MouseRegion] also expose its API, such as
+/// [InkWell.mouseCursor].
+///
+/// {@tool snippet --template=stateless_widget_material}
+/// This sample creates a rectangular region that is wrapped by a [MouseRegion]
+/// with a system mouse cursor. The mouse pointer becomes an I-beam when
+/// hovering the region.
+///
+/// ```dart imports
+/// import 'package:flutter/widgets.dart';
+/// import 'package:flutter/gestures.dart';
+/// ```
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return Center(
+///     child: MouseRegion(
+///       cursor: SystemMouseCursors.text,
+///       child: Container(
+///         width: 200,
+///         height: 100,
+///         decoration: BoxDecoration(
+///           color: Colors.blue,
+///           border: Border.all(color: Colors.yellow),
+///         ),
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// Assigning regions with mouse cursors on platforms that do not support mouse
+/// cursors, or when there are no mouse connected, will result in no-op.
+///
+/// ## Related classes
+///
+/// The following classes are designed to be created and used by the framework,
+/// therefore should not be directly used by widgets.
+///
+/// [MouseTrackerCursorMixin] is a class that manages states, and dispatches
+/// specific operations based on general mouse device updates.
+///
+/// [MouseCursorController] implements low-level imperative control by directly
+/// talking to the platform.
+///
+/// See also:
+///
+///  * [MouseRegion], which is a common way of assigning a region with a
+///    [MouseCursor].
+///  * [MouseTracker], which determines the cursor that each device should show,
+///    and dispatches the changing callbacks.
+///  * [SystemMouseCursors], which enumerates supported system cursors.
+@immutable
+abstract class MouseCursor with Diagnosticable {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const MouseCursor();
+
+  /// A very short pretty description of the mouse cursor.
+  ///
+  /// The [debugDescription] shoule be a few words that can differentiate
+  /// instances of a class to make debug information more readable. For example,
+  /// a [SystemMouseCursor] class with description "drag" will be printed as
+  /// "SystemMouseCursor(drag)".
+  ///
+  /// The [debugDescription] should not be null, but can be an empty string,
+  /// which means the class is not configurable.
```

Comment:
What do you mean by "not configurable"? Does that mean that the debugDescription can't be set?

Reply:
The class can not be set. For example, `NoopMouseCursor` can not be configured, because it takes no parameters. In this case `debugDescription` can be an empty string. 

On a second thought, this is very rare. I'm going to remove the "which means" part.